service/history/engine/testdata/engine_for_tests.go

```diff
@@ -45,9 +45,9 @@ import (
 )
 
 type EngineForTest struct {
-	Engine  engine.Engine
-	Cleanup func()
+	Engine engine.Engine
 	// Add mocks or other fields here
+	Ctx *shard.TestContext
```

Comment:
I was writing tests for refresh_workflow_tasks_test.go and I also ended up adding this guy here. A popular mock object :) I am not done anyway so will wait for your merge and rebase